var/spack/repos/builtin/packages/foam-extend/assets/spack-Allwmake

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Build wrapper script - FOAM_INST_DIR is required by foam-extend
+export FOAM_INST_DIR=$(cd .. && pwd -L)
+. $PWD/etc/bashrc ''                            # No arguments
+mkdir -p $FOAM_APPBIN $FOAM_LIBBIN 2>/dev/null  # Allow interrupt
+echo "Build openfoam with SPACK (WM_NCOMPPROCS=$WM_NCOMPPROCS, $@)"
+echo WM_PROJECT_DIR = $WM_PROJECT_DIR
+./Allwmake $@                                   # Pass arguments
+
+# Link non-dummy MPI_FOAM type to parent-dir, where rpath can find it
+if [ "${FOAM_MPI:=dummy}" != dummy -a -d "$FOAM_LIBBIN/$FOAM_MPI" ]
+then
+(
+    cd "$FOAM_LIBBIN" || exit 1
+    for i in $FOAM_MPI/lib*.so
+    do
+        [ -f $i ] && ln -sf $i "${i##*/}"
+    done
+)
+fi
+
+# -----------------------------------------------------------------------------
```

Comment:
If it makes sense, writing a `WMakePackage` is absolutely doable. All you need to do is have a new module under the `build_system` folder and collect in the base class whatever can be reused. All the OpenFOAM forks (3 by now?) will inherit from `WMakePackage` and will contain only the part that are different from one to another.